artifacts/db-reset.js

```diff
@@ -125,11 +123,17 @@ MongoClient.connect(db, (err, db) =>  {
             finalAllocations.forEach(allocation => console.log(JSON.stringify(allocation)));
 
             allocationsCol.insertMany(finalAllocations, (err, data) => {
+                if (err) {
+                    console.error(err)
+                    process.exit(1);
+                }
                 parseResponse(err, data, "allocations.insertMany");
-                console.log("Database reset performed successfully")
+                console.log("Database reset performed successfully");
                 process.exit(0);
             });
-
         });
     });
-});
+}
+
+dropDatabase()
+seedDatabase()
```

Comment:
I think this might cause regression of the Cypress test flakiness that got fixed in #214. `seedDatabase` isn't waiting for the async work queued by `dropDatabase` to finish. If `seedDatabase` manages to connect to the database faster than `dropDatabase`, then the seed data would end up deleted or incomplete.

Reply:
Agreed, good catch! The previous setup was ignoring the rejection of collections that didn't and still resolving them.
I'll call `dropDatabase()` within `seedDatabse()` or wrap them asynchronously.

Reply:
@rcowsill It didn't go as smoothly as I wanted so I decided to revert the commit. It's probably best to keep the db-reset changes out of this PR.

Reply:
> The previous setup was ignoring the rejection of collections that didn't and still resolving them.

Yes that was to avoid errors when running db-reset.js twice. The script doesn't seed the collections "contributions" or "errors", so they'd fail to delete on a second run. That isn't an issue if it's changed to drop the database, but I'm wondering if the script will fail when run on an empty MongoDB instance.

Reply:
> That isn't an issue if it's changed to drop the database, but I'm wondering if the script will fail when run on an empty MongoDB instance.

Right, that's what I noticed. It was always failing to run for the first time with the CI tests, though it ran fine manually in all cases (as the errors of dropping unknown collections weren't rejecting).

Anyways, we can always create another issue and work on it. 😄